modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -810,6 +812,12 @@ protected void paintComponent(Graphics g) {
 
         Graphics gg = null;
         try {
+            ComponentOrientation cor = this.getComponentOrientation();
+            if (!cor.equals(ComponentOrientation.UNKNOWN)) {
+                getScene().setNodeOrientation(cor.equals(ComponentOrientation.RIGHT_TO_LEFT) ?
```

Comment:
1. I think getScene() might return null (sorry, did not see this immediately)
2. if possible, could we try limit code to one statement per line, i.e.

```
Scene scene = getScene(); // it's a volatile, so better assign to a variable
if(scene != null) {
  boolean rtl = cor.equals();
  scene.setNodeOrientation(rtl ? ...);
}
```

Reply:
ok..updated..